server/src/main/java/org/jboss/as/server/deploymentoverlay/AffectedDeploymentOverlay.java

```diff
@@ -0,0 +1,212 @@
+        }
+        //Remote part
+        Operations.CompositeOperationBuilder opBuilder = Operations.CompositeOperationBuilder.create();
+        deploymentNames.forEach((deploymentName) -> opBuilder.addStep(addRedeployStep(context.getCurrentAddress().getParent().append(DEPLOYMENT, deploymentName))));
+        // Add the domain op transformer
+        List<DomainOperationTransformer> transformers = context.getAttachment(OperationAttachments.SLAVE_SERVER_OPERATION_TRANSFORMERS);
+        if (transformers == null) {
+            context.attach(OperationAttachments.SLAVE_SERVER_OPERATION_TRANSFORMERS, transformers = new ArrayList<>());
+        }
+        final ModelNode slave = opBuilder.build().getOperation();
+        transformers.add(new OverlayOperationTransformer(slave));
+    }
+
+    public static void redeployLinksAndTransformOperationForServerGroup(OperationContext context, PathAddress deploymentsRootAddress, Set<String> runtimeNames) throws OperationFailedException {
+        Set<String> deploymentNames = listDeploymentNames(context, deploymentsRootAddress, runtimeNames.stream().map(wildcardExpr -> Pattern.compile(wildcardToJavaRegex(wildcardExpr))).collect(Collectors.toSet()));
+        Operations.CompositeOperationBuilder opBuilder = Operations.CompositeOperationBuilder.create();
+        for (String deploymentName : deploymentNames) {
+            opBuilder.addStep(addRedeployStep(context.getCurrentAddress().getParent().append(DEPLOYMENT, deploymentName)));
+        }
+        // Add the domain op transformer
+        List<DomainOperationTransformer> transformers = context.getAttachment(OperationAttachments.SLAVE_SERVER_OPERATION_TRANSFORMERS);
+        if (transformers == null) {
+            context.attach(OperationAttachments.SLAVE_SERVER_OPERATION_TRANSFORMERS, transformers = new ArrayList<>());
+        }
+        final ModelNode slave = opBuilder.build().getOperation();
+        transformers.add(new OverlayOperationTransformer(slave));
+    }
+
+    /**
+     * Returns the deployment names with the specified runtime names.
+     *
+     * @param context
+     * @param runtimeNames
+     * @return
+     */
+    public static Set<String> listDeploymentNames(OperationContext context, PathAddress deploymentRootAddress, Set<Pattern> patterns) {
+        Set<String> deploymentNames = new HashSet<>();
+        Resource deploymentRootResource = context.readResourceFromRoot(deploymentRootAddress);
+        if (deploymentRootResource.hasChildren(DEPLOYMENT)) {
+            for (Resource.ResourceEntry deploymentResource : deploymentRootResource.getChildren(DEPLOYMENT)) {
+                if (isAcceptableDeployment(deploymentResource.getModel(), patterns)) {
+                    deploymentNames.add(deploymentResource.getName());
+                } else if (deploymentResource.hasChildren(SUBDEPLOYMENT)) {
```

Comment:
Does the L132 test always work correctly on an HC? There is no SUBDEPLOYMENT child on the HC.